### PR TITLE
#170 Add server configuration to Vue strings received in appConfig

### LINKED ISSUE #170: Using own vue configuration throws an error
Hi Daniel.

First of all, thank you for all your work on this.

### Description of Issue

I'm having this error, because I'm using my own Vue App strings with vue constructor in order to achieve some requirements.

### Stack Trace / Console Log
`Error: bundle export should be a function when using { runInNewContext: false }.
        at /Users/projects/moonshine2/node_modules/vue-server-renderer/build.dev.js:9381:17
        at new Promise (<anonymous>)
        at /Users/projects/moonshine2/node_modules/vue-server-renderer/build.dev.js:9368:14
        at Object.renderToString (/Users/projects/moonshine2/node_modules/vue-server-renderer/build.dev.js:9544:9)
        at Renderer.RenderToString (/Users/projects/moonshine2/node_modules/vue-pronto/lib/renderer/renderer-webpack.js:456:40)`

This happen because the file _utils/config.js_ have this code:

`if (config && config.app && config.client) {
        return {
            app: config.app,
            server: config.app,
            client: config.client,
        };
}`

On the server configuration we are assigning the app configuration, so it means a new context for Vue.

### Additional Comments

If we put 

`if (config && config.app && config.client) {
        return {
            app: config.app,
            server: config.server,
            client: config.client,
        };
}`

Works like a charm.

What do you think? Thank you in advance.

### PR DESCRIPTION
**Pull request by andoniabedul**
Related to the issue #170

### CODE REVIEW

**Comment by danielcherubini** [comment]
This is good however this introduces a breaking change. 

Could you do a check for config.server then use that else use config.app?

File: lib/utils/config.js [hunk 1]
```diff
@@ -97,14 +97,14 @@ module.exports.bootstrap = function bootstrap(serverConfig, clientConfig, mode)
...
+ * @param {{app:string, server: string, client:string}} config
  * @returns {{app:string, server:string, client:string}}
  */
 module.exports.appConfig = function appConfig(filePath, config) {
-    if (config && config.app && config.client) {
+    if (config && config.app && config.server && config.client) {
         return {
             app: config.app,
-            server: config.app,
+            server: config.server,
```
Comment (danielcherubini) [inline lib/utils/config.js]: Check first if config.server exists then use it. 

Server: config.server ? config.server : config.app

Sorry for formatting I’m on my phone.

**Review by danielcherubini — CHANGES_REQUESTED** [review]

File: lib/utils/config.js (hunk 1 above)
Comment (andoniabedul) [reply lib/utils/config.js]: No problem. Thank you for the quick review.

**Review by danielcherubini — CHANGES_REQUESTED** [review]
This is missing a unit test.. could you write one please

**Review by Zorono — APPROVED** [review]

**Comment by danielcherubini** [comment]
I’m somewhere where there’s very little internet. So I’ll look at this in a week.

**Comment by danielcherubini** [comment]
This is still missing a unit test, I can't merge it without one.

**Comment by andoniabedul** [comment]
I will do it this week

**Comment by andoniabedul** [comment]
It was added the test for server string configuration. Sorry for the delay (christmas, new year and all that pagan stuff)

**Review by danielcherubini — APPROVED** [review]

**Comment by danielcherubini** [comment]
Thanks!

**Pull request merged**